randovania/games/prime1/exporter/patch_data_factory.py

```diff
@@ -755,6 +755,26 @@ def create_game_specific_data(self) -> dict:
                 for area in region.areas:
                     level_data[region.name]["rooms"][area.name]["submerge"] = self.rng.random() < probability
 
+        # Replace vanilla missile blast shields with the new ones
+        if not self.configuration.legacy_mode:
+            for region in regions:
+                for area in region.areas:
+                    for node in area.nodes:
+                        if isinstance(node, DockNode) and node.dock_type not in elevator_dock_types:
+                            if str(node.default_dock_weakness) != "Missile Blast Shield (randomprime)":
+                                continue
+
+                            dock_node_key_value = node.extra.get("dock_index")
+                            dock_num = str(dock_node_key_value)
+                            world_name = region.name
+                            room_name = area.name
+
+                            if dock_num not in level_data[world_name]["rooms"][room_name]["doors"]:
+                                level_data[world_name]["rooms"][room_name]["doors"][dock_num] = {}
+
+                            level_data[world_name]["rooms"][room_name]["doors"][dock_num]["shieldType"] = "Blue"
+                            level_data[world_name]["rooms"][room_name]["doors"][dock_num]["blastShieldType"] = "Missile"
```

Comment:
instead of hardcoding "blue" and "missile", i'd rather use `node.default_weakness.extra["shieldType"]` and `.extra["blastShieldType"]`

Reply:
I think this might have to be hardcoded? "Blue" and "Missile" are specific to what the RP needs and is not data from the node. Unless we add this in to the logic DB nodes?

Reply:
No, this doesn't need to be hardcoded.
At this point in the code, `node` always has the randomprime missile blast shield as its default door weakness.
If you look into the RDP missile blast shield in the logic database, you will see:

```
* Missile Blast Shield (randomprime)
     Extra - shieldType: Blue
     Extra - blastShieldType: Missile
     Extra - display_name: Missile Blast Shield
```

Look, it's the exact values you hardcoded here!

Hence the above suggestion of doing `node.default_weakness.extra["shieldType"]` and `node.default_weakness.extra["blastShieldType"]`.

Reply:
Ahhh not going deep enough, just like with the node type. One day I will get the hang of this